ndcube/extra_coords/table_coord.py

```diff
@@ -10,12 +10,85 @@
+from astropy.modeling.tabular import _Tabular
 from astropy.time import Time
 from astropy.wcs.wcsapi.wrappers.sliced_wcs import combine_slices, sanitize_slices
 
 __all__ = ['TimeTableCoordinate', 'SkyCoordTableCoordinate', 'QuantityTableCoordinate']
 
 
+class _Length1Tabular(_Tabular):
+    def __init__(self, points=None, lookup_table=None, pixel_width=None,
+                 method='linear', bounds_error=True, fill_value=np.nan, **kwargs):
+        if len(lookup_table) != 1:
+            raise ValueError("lookup_table must have length 1.")
+        super().__init__(points=points, lookup_table=lookup_table, method=method,
+                         bounds_error=bounds_error, fill_value=fill_value, **kwargs)
+        self._pixel_width = pixel_width  # Width of pixel in world units.
+
+    def evaluate(self, x):
+        output = np.full(x.shape, self.fill_value)
+        output[np.logical_and(x >= -0.5, x < 0.5)] = self.lookup_table[0].value
+        return output * self.lookup_table.unit
+
+    @property
+    def inverse(self):
+        return InverseLength1Tabular(points=self.points[0], lookup_table=self.lookup_table,
+                                     pixel_width=self._pixel_width, method=self.method,
+                                     bounds_error=self.bounds_error, fill_value=self.fill_value)
+
+
+class _InverseLength1Tabular(_Length1Tabular):
+    def __init__(self, **kwargs):
+        points = kwargs.pop("points", None)
+        lookup_table = kwargs.pop("lookup_table", None)
+        super().__init__(points=lookup_table, lookup_table=points, **kwargs)
+        if self._pixel_width is None:
+            self._pixel_width = 0 * lookup_table.unit
+
+    def evaluate(self, x):
+        output = np.full(x.shape, self.fill_value)
+        diff = abs(x - self.points[0])
+        margin = self._pixel_width / 2
+        if margin.value == 0:
+            idx = diff == margin
+        else:
+            idx = np.logical_and(diff >= -1 * margin, diff < margin)
+        output[idx] = 0
+        return output * self.lookup_table.unit
+
+    @property
+    def inverse(self):
+        pass
+
+
+def length1_tabular_model(name=None, inverse=False):
+    # Adapted from astropy.modeling.tabular.tabular_model
+    dim = 1
+    table = np.zeros([2] * dim)
+    members = {'lookup_table': table, 'n_inputs': dim, 'n_outputs': 1, '_separable': True}
+
+    if inverse:
+        name_root = "InverseLength1Tabular"
+        model = _InverseLength1Tabular
+    else:
+        name_root = "Length1Tabular"
+        model = _Length1Tabular
+
+    if name is None:
+        model_id = _Tabular._id
+        _Tabular._id += 1
+        name = f'{name_root}{model_id}'
+
+    model_class = type(str(name), (model,), members)
+    model_class.__module__ = 'ndcube.extra_coords.table_coord'
+    return model_class
+
+
+Length1Tablular = length1_tabular_model(name="Length1Tabular")
+InverseLength1Tabular = length1_tabular_model(name="InverseLength1Tabular", inverse=True)
```

Comment:
We don't need this function. In Astropy it's there to dynamically generate tables for arbitary numbers of dimensions. We can just directly use the two classes you defined.

```suggestion
```